provider/pkg/await/await.go

```diff
@@ -527,6 +561,7 @@ func handleSSAIgnoreFields(c *UpdateConfig, liveOldObj *unstructured.Unstructure
 			return fmt.Errorf("unable to parse ignoreField path %q: %w", ignorePath, err)
 		}
 
+		// TODO: Enhance support for ignoreField path to support nested arrays.
```

Comment:
Ah here we go again. Might be worth lifting this comment and the split logic into one helper function so it's in one place.